src/pages/auth/sign_in/SignIn.jsx

```diff
@@ -1,32 +1,74 @@
 import "../auth.css";
 import { Link, useNavigate } from "react-router-dom";
 import { useState } from "react";
+import { useUserContext, useVideoState } from "../../../context";
+import { useDocumentTitle } from "../../../custom_hooks";
 import {
   ButtonPrimary,
   OutlineButtonPrimary,
   InputField,
   PasswordInput,
 } from "../../../components";
+import { signIn } from "../../../services";
 
 export const SignIn = () => {
   const navigate = useNavigate();
+  const [error, setError] = useState();
+  const [inputValues, setInputValues] = useState({
+    email: "",
+    password: "",
+  });
+  const guestLogin = {
+    email: "adarshbalika@gmail.com",
+    password: "adarshBalika123",
+  };
+  const { videoStateDispatch } = useVideoState();
+  const { setIsUserLoggedIn, userDataDispatch } = useUserContext();
 
   return (
     <main className="main center">
-      <form className="flex-col signup-sec">
+      <form
+        className="flex-col signup-sec"
+        onSubmit={(e) => {
+          e.preventDefault();
+          signIn({
+            setError,
+            data: inputValues,
+            videoStateDispatch,
+            userDataDispatch,
+            setIsUserLoggedIn,
+            navigate,
+          });
+        }}
+      >
         <p className="body-l">Login to my user account.</p>
 
-        <InputField label={"Email"} placeholder="Email" />
-        <PasswordInput label={"Password"} />
+        <InputField
+          value={inputValues.email}
+          onChange={(e) =>
+            setInputValues({ ...inputValues, email: e.target.value })
+          }
+          label={"Email"}
+          placeholder="Email"
+        />
+        <PasswordInput
+          value={inputValues.password}
+          onChange={(e) =>
+            setInputValues({ ...inputValues, password: e.target.value })
+          }
+          label={"Password"}
+        />
```

Comment:
I think for string there is no need of wrapping in curly braces like
label="Password"